services/redirect.ts

```diff
@@ -0,0 +1,76 @@
+import getConfig from "next/config";
+
+// https://nextjs.org/docs/api-reference/next.config.js/redirects
+export interface IRedirection {
+  source: string;
+  destination: string;
+  permanent: boolean;
+}
+
+const redirects: IRedirection[] = getConfig().serverRuntimeConfig.redirects;
+
+async function fetchLatestVersionLink(repo: string, platform: string) {
+  const fallbackVersionLink = `https://github.com/oxen-io/lokinet/releases/download/${
+    platform === "/windows"
+      ? "v0.9.7/lokinet-0.9.7-win64.exe"
+      : "v0.8.4/lokinet-v0.8.4-macos.pkg"
+  }`; // NOTE should update periodically
+
+  const res = await fetch(
+    `https://api.github.com/repos/oxen-io/${repo}/releases/latest`
+  );
+  const data = await res.json();
+  if (!data) return fallbackVersionLink;
+
+  if (res.status !== 200) {
+    console.warn(
+      `Redirect Service: Code ${res.status} | ${data.message}`,
+      `${data.documentation && `| See ${data.documentation}`}`
+    );
+    console.log(
+      `Redirect Service: Falling back to version link ${fallbackVersionLink}.`
+    );
+    return fallbackVersionLink;
+  }
+
+  let link;
+  let hasPlatformLink = data.assets.some(
+    (el: { browser_download_url: string; content_type: string }) => {
+      if (platform === "/mac") {
+        link = el.browser_download_url;
+        return (
+          el.content_type === "application/x-xar" ||
+          el.content_type === "application/octet-stream"
+        );
+      } else {
```

Comment:
platform is passed in as user input via a query paramter.
would suggest an else if windows and an else that will go to github

e.g. url given was /download?platform=solaris

Reply:
no need for this as we do not support it. See my comment above about linux redirects